tests/rbd_mirror/test_rbd_non-default_to_default_namespace_mirroring.py

```diff
@@ -0,0 +1,331 @@
+9. Initiate I/O operations on the image using rbd bench or fio or file mount:
+10. Wait till snapshot schedule interval set like above 1m
+11. Verify that data is mirrored from the primary to the secondary cluster
+12. Verify data consistency using md5sum checksum from primary and secondary
+13. Similarly, create image in the default namespace in cluster2 and verify that the image
+    got mirrored to cluster1 remote namespace
+14. Add a snapshot schedule for the mirrored image in the namespace
+15. Initiate I/O operations on the image using rbd bench, fio, or file mount
+16. Wait till snapshot schedule interval set like above 1m.
+17. Verify that data is mirrored from cluster-2 to cluster-1
+18. Verify data consistency using md5sum checksum from primary and secondary
+19.  Repeat the above test on EC pool
+20. Cleanup the images, namespace, pools along with disk cleanup.
+"""
+
+import ast
+import time
+
+from ceph.rbd.initial_config import initial_mirror_config
+from ceph.rbd.utils import check_data_integrity, getdict, random_string
+from ceph.rbd.workflows.cleanup import cleanup
+from ceph.rbd.workflows.krbd_io_handler import krbd_io_handler
+from ceph.rbd.workflows.rbd_mirror import enable_image_mirroring, wait_for_status
+from ceph.rbd.workflows.snap_scheduling import (
+    add_snapshot_scheduling,
+    verify_namespace_snapshot_schedule,
+    verify_snapshot_schedule,
+)
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def test_non_default_to_default_namespace_mirroring(
+    pri_config, sec_config, pool_types, **kw
+):
+    log.info(
+        "Starting CEPH-83612860 - Snapshot based Namespace "
+        + "level mirroring from Non default to default namespace"
+    )
+    rbd_primary = pri_config.get("rbd")
+    rbd_secondary = sec_config.get("rbd")
+    client_primary = pri_config.get("client")
+    client_secondary = sec_config.get("client")
+
+    for pool_type in pool_types:
+        rbd_config = kw.get("config", {}).get(pool_type, {})
+        multi_pool_config = getdict(rbd_config)
+
+        for pool, pool_config in multi_pool_config.items():
+            multi_image_config = getdict(pool_config)
+            image_config = {k: v for k, v in multi_image_config.items()}
+            namespace = pool_config.get("namespace")
+            remote_namespace = pool_config.get("remote_namespace")
+
+            for image, image_config_val in image_config.items():
+                image_enable_config = {
+                    "pool": pool,
+                    "image": image,
+                    "mirrormode": "snapshot",
+                    "namespace": namespace,
+                    "remote_namespace": remote_namespace,
+                }
+                # Enable snapshot mode mirroring on images of the namespace
+                enable_image_mirroring(pri_config, sec_config, **image_enable_config)
+                # Verify image mirroring status on primary cluster
```

Comment:
please add `mirror image status` and `pool status` output in `log.info` before snapshot schedule and after snapshot schedule.